langchain/tools/base.py

```diff
@@ -2,241 +2,68 @@
+
+    # Child tools can add additional validation by
+    # subclassing this schema.
+    tool_input: str
+
+
+class BaseTool(ABC, BaseStructuredTool[str]):
     """Interface LangChain tools must implement."""
 
-    name: str
-    description: str
-    args_schema: Optional[Type[BaseModel]] = None
-    """Pydantic model class to validate and parse the tool's input arguments."""
-    return_direct: bool = False
-    verbose: bool = False
-    callback_manager: BaseCallbackManager = Field(default_factory=get_callback_manager)
-
-    class Config:
-        """Configuration for this pydantic object."""
-
-        extra = Extra.forbid
-        arbitrary_types_allowed = True
-
-    @property
-    def args(self) -> dict:
-        if self.args_schema is not None:
-            return self.args_schema.schema()["properties"]
-        else:
-            inferred_model = validate_arguments(self._run).model  # type: ignore
-            return get_filtered_args(inferred_model, self._run)
-
-    def _parse_input(
-        self,
-        tool_input: Union[str, Dict],
-    ) -> None:
-        """Convert tool input to pydantic model."""
-        input_args = self.args_schema
-        if isinstance(tool_input, str):
-            if input_args is not None:
-                key_ = next(iter(input_args.__fields__.keys()))
-                input_args.validate({key_: tool_input})
-        else:
-            if input_args is not None:
-                input_args.validate(tool_input)
-
-    @validator("callback_manager", pre=True, always=True)
-    def set_callback_manager(
-        cls, callback_manager: Optional[BaseCallbackManager]
-    ) -> BaseCallbackManager:
-        """If callback manager is None, set it.
-
-        This allows users to pass in None as callback manager, which is a nice UX.
-        """
-        return callback_manager or get_callback_manager()
+    args_schema: Type[StringSchema] = StringSchema  # :meta private:
+
+    def _wrap_input(self, tool_input: Union[str, Dict]) -> Dict:
+        """Wrap the tool's input into a pydantic model."""
+        if isinstance(tool_input, dict):
+            return tool_input
+        return {"tool_input": tool_input}
+
+    def _prepare_input(self, input_: dict) -> Tuple[Sequence, Dict]:
+        """Prepare the args and kwargs for the tool."""
+        # We expect a single string input
+        return tuple(input_.values()), {}
 
     @abstractmethod
-    def _run(self, *args: Any, **kwargs: Any) -> str:
+    def _run(self, tool_input: str) -> str:
         """Use the tool."""
 
     @abstractmethod
-    async def _arun(self, *args: Any, **kwargs: Any) -> str:
+    async def _arun(self, tool_input: str) -> str:
         """Use the tool asynchronously."""
 
     def run(
         self,
-        tool_input: Union[str, Dict],
-        verbose: Optional[bool] = None,
-        start_color: Optional[str] = "green",
-        color: Optional[str] = "green",
+        tool_input: Union[str, dict],
```

Comment:
To satisfy linting / maintain proper inheritance, I can't strictly type this as `str`. We could introduce another generic, but since python didn't introduce default generics until > 3.10, I think it would require a triangle relationship that seems undesireable.  The default generic would be a prereq to mimicing the typescript implementation